org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -218,11 +215,13 @@ private String create() throws Exception {
 						+ " INNER JOIN M_Product p ON (p.M_Product_ID=po.M_Product_ID)"
 						+ " INNER JOIN M_DiscountSchemaLine dl ON (dl.M_DiscountSchemaLine_ID=?) "	//	#2
 						+ "WHERE p.AD_Client_ID IN (?, 0)"		//	#3
-						+ " AND p.IsActive='Y' AND po.IsActive='Y' AND po.IsCurrentVendor='Y'"
+						+ " AND p.IsActive='Y' AND po.IsActive='Y'"
 						//	Optional Restrictions
 						+ " AND (dl.Group1 IS NULL OR p.Group1=dl.Group1)"
 						+ " AND (dl.Group2 IS NULL OR p.Group2=dl.Group2)"
-						+ " AND (dl.C_BPartner_ID IS NULL OR po.C_BPartner_ID=dl.C_BPartner_ID)"
+						+ " AND (dl.VendorCategory IS NULL OR po.VendorCategory=dl.VendorCategory)"
+						+ " AND (dl.C_BPartner_ID IS NULL AND po.IsCurrentVendor='Y'"

```

Comment:
please use parenthesis to make this easier to read and remove any potential ambiguaties

Reply:
Yes, it's better.